kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -64,12 +97,24 @@ struct KinematicsQueryOptions
 {
   KinematicsQueryOptions() :
     lock_redundant_joints(false),
-    return_approximate_solution(false)
+    return_approximate_solution(false),
+  	solutions_search_code(KinematicSearches::ONE)
   {
   }
 
   bool lock_redundant_joints;
   bool return_approximate_solution;
+  KinematicSearch solutions_search_code;  // used when finding multiple solutions for the same pose
+};
+
+/*
+ * @struct KinematicsResult
+ * @brief Reports result details of an ik solution query
+ */
+struct KinematicsResult
+{
+	KinematicError kinematic_error;
+	double solution_percentage; // the percentage of solutions achieved over the total number of solutions searched.
```

Comment:
Should be a doxygen comment.  I don't think these members can be commented enough.  I would suggest that more comments with examples are needed.